bundler/spec/install/gemfile/specific_platform_spec.rb

```diff
@@ -1262,6 +1262,80 @@
     end
   end
 
+  it "adds current musl platform" do
+    build_repo4 do
+      build_gem "rcee_precompiled", "0.5.0" do |s|
+        s.platform = "x86_64-linux"
+      end
+
+      build_gem "rcee_precompiled", "0.5.0" do |s|
+        s.platform = "x86_64-linux-musl"
+      end
+    end
+
+    gemfile <<~G
+      source "#{file_uri_for(gem_repo4)}"
+
+      gem "rcee_precompiled", "0.5.0"
+    G
+
+    simulate_platform "x86_64-linux-musl" do
+      bundle "lock", artifice: "compact_index", env: { "BUNDLER_SPEC_GEM_REPO" => gem_repo4.to_s }
+
+      expect(lockfile).to eq(<<~L)
+        GEM
+          remote: #{file_uri_for(gem_repo4)}/
+          specs:
+            rcee_precompiled (0.5.0-x86_64-linux)
+            rcee_precompiled (0.5.0-x86_64-linux-musl)
+
+        PLATFORMS
+          x86_64-linux
+          x86_64-linux-musl
+
+        DEPENDENCIES
+          rcee_precompiled (= 0.5.0)
+
+        BUNDLED WITH
+           #{Bundler::VERSION}
+      L
+    end
+  end
+
+  it "does not add current platform if there's a more specific one already locked" do
```

Comment:
this looks like a _less_ specific platform is locked?

Reply:
Yeah, I did not explain this case correctly. In order to avoid too many platforms from being locked (case of `x86_64-darwin-18`, `x86_64-darwin-19`, `x86_64-darwin-20`, etc), I made it so that if one of the resolved platforms (say, `x86_64-darwin`) is compatible with the current platform but less specific, then we we'll keep that and don't add the current specific platform. I will reword it to make it clear!